Adopt some more smart pointers in UIProcess

annevk opened the pull request:
#### cebc57a827c83b72b05799c50c38f3a8bb2a860f
<pre>
Adopt some more smart pointers in UIProcess
<a href="https://bugs.webkit.org/show_bug.cgi?id=295173">https://bugs.webkit.org/show_bug.cgi?id=295173</a>

Reviewed by NOBODY (OOPS!).

Apply <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines</a>

* Source/WebKit/UIProcess/XR/openxr/*

GTK unified build fixes by Sergio Villar.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cebc57a827c83b72b05799c50c38f3a8bb2a860f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/109844 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/29502 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/19932 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/115865 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/60081 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30180 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38090 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/83498 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/60081 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/112792 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/24059 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/98920 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/63941 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/23438 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/59660 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/93429 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/17111 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/118657 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/36883 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/27332 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/92479 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37256 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/95187 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92302 "Passed tests") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/37278 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/15021 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/32729 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17722 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/36778 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42248 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/36438 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/39780 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/38147 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b401154185d4314e77ccc629b1c0febace7353cb. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47351

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1550f3d4b434026afaf8e56d89e803286840ebe5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47351

annevk commented on Source/WebKit/UIProcess/PageClient.h:
```diff
@@ -610,7 +610,7 @@ class PageClient : public CanMakeWeakPtr<PageClient> {
 
     // Auxiliary Client Creation
 #if ENABLE(FULLSCREEN_API)
-    virtual WebFullScreenManagerProxyClient& fullScreenManagerProxyClient() = 0;
```
All instances want the checked variant. When we've had that elsewhere we haven't added a wrapper.

cdumez replied:
And now people are force to use a checked variant even if they don't need one. We should not force it on people. The pattern is always to provide alternative protected/checked getters.

annevk replied:
Is this different from https://github.com/WebKit/WebKit/pull/47174#pullrequestreview-2962524254 because these are public? Anyway, happy to add wrappers.

annevk replied:
Oh, I thought you were not a fan because you did not notice the name change. Well, good to know and thanks for reviewing!

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6cbd85ab321784c34cb1e766b312f34004e71b73. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47351

annevk commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -1856,7 +1856,7 @@ void WebPageProxy::close()
     RunLoop::currentSingleton().dispatch([processesToClose = WTFMove(processesToClose)] {
         for (auto [process, pageID, scope] : processesToClose)
-            process->send(Messages::WebPage::Close(), pageID);
+            Ref { process }->send(Messages::WebPage::Close(), pageID);
```
This does not appear to have worked unfortunately. Maybe due to auto. I'll file a radar and address this for now as it's not too bad I suspect.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/13e02c8cfbd1527077b6e897f9c4c6f05fc06e39. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47351

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/71faf39ee9ee9cc59c96c353678852d2f9761e21. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47351

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/cebc57a827c83b72b05799c50c38f3a8bb2a860f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47351

annevk committed:
Adopt some more smart pointers in UIProcess
https://bugs.webkit.org/show_bug.cgi?id=295173

Reviewed by Charlie Wolfe.

Apply https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines

* Source/WebKit/UIProcess/XR/openxr/*

GTK unified build fixes by Sergio Villar.

Canonical link: https://commits.webkit.org/296867@main
